### PR TITLE
Add antialiasing toggle to text tool

### PR DESCRIPTION
**Pull request by potatoes1286**
I was frustrated with the lack of ability to disable antialiasing on Pinta, so I decided to just go and add it.

Please note this was cobbled together. I have zero experience with Cairo/Pango/PangoCairo but in my brief testing there didn't seem to be any obvious problems.

Thank you in advance.

### CODE REVIEW

**Comment by cameronwhite** [comment]
Thanks for having a look! I'll need to have a look at the pango / Cairo stuff and do some testing, but for the toolbar button / settings I think it would be best to reuse the standard antialiasing button from the base class ( `BaseTool`). I think you just need to override `ShowAntialiasingButton` and then check the `UseAntialiasing` property when drawing

**Comment by potatoes1286** [comment]
> Thanks for having a look! I'll need to have a look at the pango / Cairo stuff and do some testing, but for the toolbar button / settings I think it would be best to reuse the standard antialiasing button from the base class ( `BaseTool`). I think you just need to override `ShowAntialiasingButton` and then check the `UseAntialiasing` property when drawing

This is possible, but as redraw is only called when a change is made, changing the anti-aliasing setting before text finalization does not kick in until another action is performed (e.g moving cursor, adding new text to text, etc.)

**Comment by cameronwhite** [comment]
Right, that makes sense. I think it would be reasonable to add an event in BaseTool for this so that re-editable tools can handle the antialiasing setting changing and trigger their own redraw.

**Comment by potatoes1286** [comment]
@cameronwhite Added this functionality to BaseTool, and readjusted the text tool to support it.

File: Pinta.Core/Classes/BaseTool.cs [hunk 1]
```diff
@@ -303,6 +303,14 @@ protected virtual void OnSaveSettings (ISettingsService settings)
 			settings.PutSetting (ANTIALIAS_SETTING, antialiasing_button.SelectedIndex);
 	}
 
+	/// <summary>
+	/// Called when the tool is deselected from the toolbox.
+	/// </summary>
+	protected virtual void OnChangeAntialias (object? sender, EventArgs e)
```
Comment (cameronwhite) [inline Pinta.Core/Classes/BaseTool.cs:309]: To match some of the other methods' naming, I'd suggest `OnAntaliasingChanged`, and I'd also remove the `sender` / `e` arguments since they aren't useful for any of the derived classes that might override this? 
Also, it looks like the documentation comment might have been copied from elsewhere?

File: Pinta.Tools/Tools/TextTool.cs [hunk 2]
```diff
@@ -413,6 +417,11 @@ private void HandleSelectedLayerChanged (object? sender, EventArgs e)
 		UpdateFont ();
 	}
 
+	protected override void OnChangeAntialias (object? sender, EventArgs e)
+	{
+		UpdateFont();
```
Comment (cameronwhite) [inline Pinta.Tools/Tools/TextTool.cs:422]: I noticed that after I finished typing some text (i.e. after pressing `Esc`) that the antialiasing changes didn't cause a redraw right away until I did something else to force a redraw, like pressing Ctrl.

This also happens with other font properties too, though, so it's a more general behaviour and could probably be left for another pull request

File: Pinta.Tools/Tools/TextTool.cs [hunk 3]
```diff
@@ -975,7 +984,35 @@ private void RedrawText (bool showCursor, bool useTextLayer)
 			ClearTextLayer ();
 		}
 
+		/*		var antialiasing = antialiasing_button.SelectedItem.GetTagOrDefault (true);
```
Comment (cameronwhite) [inline Pinta.Tools/Tools/TextTool.cs:987]: This old code can be removed

File: Pinta.Tools/Tools/TextTool.cs [hunk 4]
```diff
@@ -975,7 +984,35 @@ private void RedrawText (bool showCursor, bool useTextLayer)
...
+			var options = new Cairo.FontOptions ();
+			options.Antialias = Antialias.Gray;
+			PangoCairo.Functions.ContextSetFontOptions (PintaCore.Chrome.MainWindow.GetPangoContext (), options);
+		}
+		else
+		{
+			g.Antialias = Cairo.Antialias.None;
+			var options = new Cairo.FontOptions ();
+			options.Antialias = Antialias.None;
+			PangoCairo.Functions.ContextSetFontOptions (PintaCore.Chrome.MainWindow.GetPangoContext (), options);
```
Comment (cameronwhite) [inline Pinta.Tools/Tools/TextTool.cs:1013]: A minor refactoring would be to extract `options` and the `ContextSetFontOptions` call outside the `if` block to avoid the duplication in the two cases

File: Pinta.Core/Classes/BaseTool.cs (hunk 1 above)
Comment (potatoes1286) [reply Pinta.Core/Classes/BaseTool.cs]: Oops. Fixed both the documentation and naming convention, thanks.

File: Pinta.Tools/Tools/TextTool.cs (hunk 2 above)
Comment (potatoes1286) [reply Pinta.Tools/Tools/TextTool.cs:422]: Yes, it seems like in that state after pressing escape, the text is not considered in edit mode, and so does not call redraw.

That edit flag seems to be there for a reason (starting without it with text tool on causes crash), so i won't touch it for the time being in case it causes other issues.

File: Pinta.Tools/Tools/TextTool.cs (hunk 3 above)
Comment (potatoes1286) [reply Pinta.Tools/Tools/TextTool.cs]: Removed. Thanks for pointing that out.

File: Pinta.Tools/Tools/TextTool.cs (hunk 4 above)
Comment (potatoes1286) [reply Pinta.Tools/Tools/TextTool.cs]: Done

**Comment by cameronwhite** [comment]
The changes look good to me - thanks!
The CI build is just failing from the code formatting check - you should be able to run `dotnet format` to fix that
